doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,229 @@
+    - ***(Single-point)***
+    - Classify a single point and compute class probabilities.
+    - The predicted class is stored in `prediction`.
+
+---
+
+ * `tree.Classify(data, predictions)`
+    - ***(Multi-point)***
+    - Classify a set of points.
+    - The prediction for data point `i` can be accessed with `predictions[i]`.
+
+---
+
+#### Classification Parameters:
+
+| **usage** | **name** | **type** | **description** |
+|-----------|----------|----------|-----------------|
+| _single-point_ | `point` | [`arma::vec`](../matrices.md) | Single point for classification. |
+| _single-point_ | `prediction` | `size_t&` | `size_t` to store class prediction into. |
+||||
+| _multi-point_ | `data` | [`arma::mat`](../matrices.md) | Set of [column-major](../matrices.md#representing-data-in-mlpack) points for classification. |
+| _multi-point_ | `predictions` | [`arma::Row<size_t>&`](../matrices.md) | Vector of `size_t`s to store class prediction into.  Will be set to length `data.n_cols`. |
+
+***Note:*** different types can be used for `data` and `point` (e.g.
+`arma::fmat`, `arma::sp_mat`, `arma::sp_vec`, etc.).  However, the element type
+that is used should be the same type that was used for training.
+
+### Other Functionality
+
+ * `WeakLearner(size_t i)` returns a `WeakLearnerType` indicating which weak learner 
+    is being used. 
+
+ * `gb.NumModels()` returns a `size_t` indicating how many weak learners are being
+    used by the model.
+
+ * `gb.NumClasses()` returns a `size_t` indicating the number of classes the
+   model was trained on.
+
+For complete functionality, the [source
+code](/src/mlpack/methods/grad_boosting/grad_boosting.hpp) can be consulted.
+Each method is fully documented.
+
+### Simple Examples
+
+See also the [simple usage example](#simple-usage-example) for a trivial use of
+`GradBoosting`.
+
+---
+
+Train a decision tree on mixed categorical data and save it:
+
+```c++
+// Load a categorical dataset.
+arma::mat dataset;
+mlpack::data::DatasetInfo info;
+// See https://datasets.mlpack.org/covertype.train.arff.
+mlpack::data::Load("covertype.train.arff", dataset, info, true);
+
+arma::Row<size_t> labels;
+// See https://datasets.mlpack.org/covertype.train.labels.csv.
+mlpack::data::Load("covertype.train.labels.csv", labels, true);
+
+// Create the tree.
+mlpack::GradBoosting gb;
+// Train on the given dataset, specifying number of weak learners at 5.
+gb.Train(dataset, labels, 7 /* classes */, 5 /* number of weak learners */);
+
+// Load categorical test data.
+arma::mat testDataset;
+// See https://datasets.mlpack.org/covertype.test.arff.
+mlpack::data::Load("covertype.test.arff", testDataset, info, true);
+
+// Predict class of first test point.
+const size_t firstPrediction = gb.Classify(testDataset.col(0));
+std::cout << "Predicted class of first test point is " << firstPrediction << "."
+    << std::endl;
+
+// Save the tree to `tree.bin`.
+mlpack::data::Save("gb.bin", "gb", gb);
+```
+
+---
+
```

Comment:
It would be necessary I would say to show an example showing how this GradientBoosting can be used with different types of `WeakLearners`, and at least one with floating point as well.